File: app/routes/stats.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from ..db.session import get_db
from ..services.stats_service import StatsService
import logging

logger = logging.getLogger(__name__)

router = APIRouter()

@router.get("/dashboard")
def get_dashboard_stats(db: Session = Depends(get_db)):
    """
    Récupère toutes les statistiques pour le tableau de bord
    """
    try:
        stats_service = StatsService(db)
        return stats_service.get_dashboard_stats()
    except Exception as e:
        logger.error(f"Erreur lors de la récupération des statistiques du tableau de bord: {str(e)}")
        raise HTTPException(
            status_code=500,
            detail="Erreur lors de la récupération des statistiques du tableau de bord"
        ) 

File: src/routes/stats.ts
import express from "express";
import { StatsController } from "../controllers/StatsController";

const router = express.Router();
const statsController = new StatsController();

// Route pour obtenir toutes les statistiques du tableau de bord
router.get("/dashboard", statsController.getDashboardStats);

// Route pour obtenir les statistiques quotidiennes
router.get("/daily", statsController.getDailyStats);

// Route pour obtenir les statistiques par type
router.get("/types", statsController.getTypeStats);

// Route pour obtenir les statistiques géographiques
router.get("/geographic", statsController.getGeographicStats);

export default router;
